tests/merge.test.ts

```diff
@@ -0,0 +1,118 @@
+import { EventPayload } from '../src/EventPayload';
+import { merge } from '../src/merge';
+
+describe('Merge Function Test', () => {
+  it('should merge primitives values correctly', () => {
+    const event1: EventPayload = {
+      action: 'APPLY',
+      bot: false,
+      locale: 'en_US',
+    };
+
+    const uneffectedEvent1: EventPayload = {...event1};
+
+    const event2: Record<string, unknown> = {
+      label: 'label',
+      bot: null,
+      locale: 'en_UK',
+    };
+
+    const result = merge(event1, event2);
+
+    const expected: EventPayload = {
+      label: 'label',
+      locale: 'en_UK',
+      action: 'APPLY'
+    };
+
+    expect(result).toEqual(expected);
+    expect(event1).toEqual(uneffectedEvent1); // confirm event1 uneffected
+  });
+
+  it('should merge objects with no overlap correctly', () => {
+    const event1: EventPayload = {
+      action: 'APPLY',
+      chat: {
+        botId: 'botId',
+      }
+    };
+
+    const uneffectedEvent1: EventPayload = {...event1};
+
+    const event2: Record<string, unknown> = {
+      action: 'APPLY',
+      browserAgent: {
+        device: 'iPhone',
+        deviceClass: 'Mobile',
+      },
+      chat: {
+        responseId: 'responseId',
+      }
+    };
+
+    const result = merge(event1, event2);
+
+    const expected: EventPayload = {
+      action: 'APPLY',
+      browserAgent: {
+        device: 'iPhone',
+        deviceClass: 'Mobile',
+      },
+      chat: {
+        botId: 'botId',
+        responseId: 'responseId',
+      },
+    };
+
+    expect(result).toEqual(expected);
+    expect(event1).toEqual(uneffectedEvent1); // confirm event1 uneffected
+  });
+
+  it('should merge primitives and objects (with overlap) correctly', () => {
```

Comment:
We should also add a test to cover merging with values not in the EventPayload.

Reply:
Discussed offline. This validation is not really possible without having a hardcoded data structure of the possible keys and comparing that to what is in the record. This is a complicated process and not really necessary considering that the request will eventually be sent to the Events API, which will throw an error if the body is invalid (and will describe what is invalid about it). 

However, by changing the type of the record to be Partial<Record<keyof EventPayload, unknown>> we can provide some type checking in the IDE if the input is not valid for EventPayload (likely only for typescript users).